health-services/PassiveGoals/app/src/main/java/com/example/passivegoals/HealthServicesManager.kt

```diff
@@ -64,26 +58,43 @@ class HealthServicesManager @Inject constructor(
     private val passiveMonitoringClient = healthServicesClient.passiveMonitoringClient
 
     suspend fun hasFloorsAndDailyStepsCapability(): Boolean {
-        val capabilities = passiveMonitoringClient.capabilities.await()
-        return capabilities.supportedDataTypesEvents.containsAll(
+        val capabilities = passiveMonitoringClient.getCapabilitiesAsync().await()
+        return capabilities.supportedDataTypesPassiveGoals.containsAll(
             setOf(
-                DataType.TOTAL_CALORIES,
-                DataType.FLOORS
+                DataType.STEPS_DAILY,
+                DataType.FLOORS_DAILY,
             )
         )
     }
 
-    suspend fun subscribeForGoals() {
+    @OptIn(ExperimentalCoroutinesApi::class)
+    suspend fun subscribeForGoals(): Boolean {
         Log.i(TAG, "Subscribing for goals")
-        val componentName = ComponentName(context, PassiveGoalsReceiver::class.java)
-        // Each goal is a separate subscription.
-        passiveMonitoringClient.registerPassiveGoalCallback(dailyStepsGoal, componentName).await()
-        passiveMonitoringClient.registerPassiveGoalCallback(floorsGoal, componentName).await()
+        val passiveListenerConfig =
+            PassiveListenerConfig.builder().setDailyGoals(setOf(dailyStepsGoal, floorsGoal))
+                .build()
+        val future = passiveMonitoringClient.setPassiveListenerServiceAsync(
```

Comment:
Use await()?

Reply:
`setPassiveListenerServiceAsync` returns a `ListenableFuture<Void>` (so `.async()` would also return `Void`). So I did things this way (returning true on success) so that in `MainViewModel` you can do:

```kotlin
if (!healthServicesManager.subscribeForGoals()) {
    _uiState.value = UiState.CapabilitiesNotAvailable
}
```

Not 100% sure if this is the right way to go about it. It's a lot of boilerplate code, for sure.

Reply:
Oh, I had hope that setPassiveListenerServiceAsync would throw if not successful?

so

```
try {
    setPassiveListenerServiceAsync(...).await()
} catch {
}
```

Maybe it doesn't though? Just trying to think of ways to completely avoid Futures and callbacks...

Reply:
Ah! I need to do some testing. But to check my understanding, you're suggesting this?

```kotlin
        return try {
            passiveMonitoringClient.setPassiveListenerServiceAsync(
                PassiveGoalService::class.java,
                passiveListenerConfig
            ).await()
            true
        } catch (t: Throwable) {
            false
        }
```

(The boolean return is needed so we can set the UI state to `UiState.CapabilitiesNotAvailable`, as above. Not the greatest architecture but the easiest thing to do without making more widespread changes.)